### PR TITLE
Update .NET SDK with Traffic Manager 2018-03-01 API version.

### PR DESCRIPTION
**Pull request by ttlcgl**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by tatlicioglu** [comment]
Spec change: https://github.com/Azure/azure-rest-api-specs/commit/7a65dadde5c7a77412003f30ee29fd9924b5497b#diff-e142ce0a89cbb105e0cb3f49055fcc59

**Review by dsgouda — CHANGES_REQUESTED** [review]
- Please create a generate.ps1 file similar to [this]() and regenerate the code.
- Please run `msbuild build.proj /t:build /p:Scope=SDKs\TrafficManager` 
- Please commit any files that may be modified.
- Please rebase all commits into a single commit and update the PR
- There are breaking changes being introduced here. 
- Please update the package version to `2.4.0-preview`

**Comment by tatlicioglu** [comment]
I've addressed remarks.

File: src/SDKs/TrafficManager/Management.TrafficManager/Microsoft.Azure.Management.TrafficManager.csproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
 		<PackageId>Microsoft.Azure.Management.TrafficManager</PackageId>
 		<Description>Microsoft Azure Management TrafficManager Library</Description>
 		<AssemblyName>Microsoft.Azure.Management.TrafficManager</AssemblyName>
-		<Version>2.3.0-preview</Version>
+		<Version>2.4.0-preview</Version>
 		<PackageTags>Microsoft Azure TrafficManager management;TrafficManager;TrafficManager management;</PackageTags>
 		<PackageReleaseNotes/>
```
Comment (dsgouda) [inline src/SDKs/TrafficManager/Management.TrafficManager/Microsoft.Azure.Management.TrafficManager.csproj]: Please add `PackageReleaseNotes` similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/33056293d935f23f00595de23dbd8799ea852efb/src/AzureStack/Management/Subscription/Subscription/Microsoft.AzureStack.Management.Subscription.csproj#L10).

File: src/SDKs/TrafficManager/TrafficManager.Tests/TrafficManager.Tests.csproj [hunk 2]
```diff
@@ -10,7 +10,7 @@
     <TargetFrameworks>netcoreapp1.1</TargetFrameworks>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.6.0-preview" />
```
Comment (dsgouda) [inline src/SDKs/TrafficManager/TrafficManager.Tests/TrafficManager.Tests.csproj]: Please confirm tests can run without this package reference.

File: src/SDKs/TrafficManager/TrafficManager.Tests/TrafficManager.Tests.xproj [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (dsgouda) [inline src/SDKs/TrafficManager/TrafficManager.Tests/TrafficManager.Tests.xproj]: @shahabhijeet do we allow xproj files?

File: src/SDKs/TrafficManager/TrafficManager.Tests/project.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{
+  "version": "1.0.0-*",
```
Comment (dsgouda) [inline src/SDKs/TrafficManager/TrafficManager.Tests/project.json]: I believe we do not allow project.json and xproj files.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please make sure the sdk is generated using the generate.ps1 script. This should generate a .txt and a SdkInfo_*.cs file. Please commit these files.
Also when you run `msbuild build.proj /t:build /p:Scope=SDKs\TrafficManager` [this](https://github.com/tatlicioglu/azure-sdk-for-net/blob/45458bf353219cb4e205b80b64e197ee6d25133c/src/SDKs/TrafficManager/AzSdk.RP.props) file should be modified.

File: src/SDKs/TrafficManager/TrafficManager.Tests/TrafficManager.Tests.csproj (hunk 2 above)
Comment (tatlicioglu) [reply src/SDKs/TrafficManager/TrafficManager.Tests/TrafficManager.Tests.csproj]: Yes, they work fine. We are not using this package anymore. I've already made required changes in test code.

File: src/SDKs/TrafficManager/TrafficManager.Tests/TrafficManager.Tests.xproj (hunk 3 above)
Comment (tatlicioglu) [reply src/SDKs/TrafficManager/TrafficManager.Tests/TrafficManager.Tests.xproj]: Removed.

File: src/SDKs/TrafficManager/TrafficManager.Tests/project.json (hunk 4 above)
Comment (tatlicioglu) [reply src/SDKs/TrafficManager/TrafficManager.Tests/project.json]: Removed both.

File: src/SDKs/TrafficManager/Management.TrafficManager/Microsoft.Azure.Management.TrafficManager.csproj (hunk 1 above)
Comment (tatlicioglu) [reply src/SDKs/TrafficManager/Management.TrafficManager/Microsoft.Azure.Management.TrafficManager.csproj]: Done.

**Comment by tatlicioglu** [comment]
Thanks @dsgouda,

I've addressed remarks.

File: src/SDKs/_metadata/Microsoft_Networking.txt [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿Executing AutoRest command
```
Comment (dsgouda) [inline src/SDKs/_metadata/Microsoft_Networking.txt]: I don't think this file was generated.

File: src/SDKs/TrafficManager/TrafficManager.Tests/ScenarioTests/EndpointScenarioTests.Features.cs [hunk 6]
```diff
@@ -1,70 +0,0 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
-
-namespace Microsoft.Azure.Management.TrafficManager.Testing.ScenarioTests
-{
-    using System.Collections.Generic;
```
Comment (dsgouda) [inline src/SDKs/TrafficManager/TrafficManager.Tests/ScenarioTests/EndpointScenarioTests.Features.cs]: Do you plan to delete these tests forever. They could be skipped if you intend to enable them at a future point.

**Review by dsgouda — CHANGES_REQUESTED** [review]
`generate.ps1` has `ResourceProvider` as `Microsoft/Networking` which should instead be the name of this RP
Looks good other than that. Thanks for the quick turnaround.

File: src/SDKs/TrafficManager/TrafficManager.Tests/ScenarioTests/EndpointScenarioTests.Features.cs (hunk 6 above)
Comment (tatlicioglu) [reply src/SDKs/TrafficManager/TrafficManager.Tests/ScenarioTests/EndpointScenarioTests.Features.cs]: These are mostly duplicate of EndpointScnearioTests.cs. We are removing them permanently. They are not adding any extra coverage, but causing build failure due to partial definition of classes.

File: src/SDKs/_metadata/network_resource-manager.txt [hunk 7]
```diff
@@ -1,17 +0,0 @@
-﻿Executing AutoRest command
-cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=D:\Projects\azure-sdk-for-net\tools\..\src\SDKs\
-2018-05-31 00:48:46 UTC
-1) azure-rest-api-specs repository information
-GitHub fork: Azure
-Branch:      master
-Commit:      a5af2bc46e91b4b6f6aa25408c1814532fcfc041
-
-2) AutoRest information
```
Comment (dsgouda) [inline src/SDKs/_metadata/network_resource-manager.txt]: Don't delete, check this out from the remote branch.

**Review by dsgouda — APPROVED** [review]
LGTM, simply undo changes to the txt file.

File: src/SDKs/_metadata/network_resource-manager.txt (hunk 7 above)
Comment (tatlicioglu) [reply src/SDKs/_metadata/network_resource-manager.txt]: Good catch, thanks. Done.

**Comment by tatlicioglu** [comment]
@dsgouda  Who can merge this? I am not authorized.

**Comment by dsgouda** [comment]
@tatlicioglu will merge once CIs pass

**Pull request merged**